packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -325,6 +346,8 @@ class _MonthPickerState extends State<MonthPicker> {
   @override
   void initState() {
     super.initState();
+    // Initially display the pre-selected date.
+    _currentDisplayedMonthDate = new DateTime(config.selectedDate.year, config.selectedDate.month);
```

Comment:
You need to update this member in `didUpdateConfig` as well.  When your parent builds a new widget for this location in the tree and this `State` object is re-used, the framework calls `didUpdateConfig` to let you know that your `config` property now points to the new widget.